File: shared/src/commonMain/kotlin/com/ttcoachai/shared/drill/MetricPrecision.kt
package com.ttcoachai.shared.drill

/**
 * Trust rule (context doc §3): precise degrees ONLY for in-plane metrics;
 * everything else is qualitative-only or silent.
 */
enum class MetricPrecision { PRECISE_DEGREES, QUALITATIVE }

object MetricPrecisionPolicy {

    /**
     * Trust rule (context doc §3): precise degrees ONLY for these in-plane angle metrics.
     * Explicitly enumerated — NOT derived from [DrillMetrics.ALL_KEYS], because the derived
     * keys stroke_speed and coil_ratio are qualitative proxies, not degree measurements.
     */
    private val preciseKeys = setOf(
        DrillMetrics.METRIC_ELBOW_ANGLE,
        DrillMetrics.METRIC_SHOULDER_ANGLE,
        DrillMetrics.METRIC_KNEE_BEND,
        DrillMetrics.METRIC_TORSO_LEAN,
        DrillMetrics.METRIC_FOLLOW_THROUGH_ANGLE_2D,
    )

    /** Unknown metrics default to QUALITATIVE — never overclaim precision. */
    fun precisionFor(metricKey: String): MetricPrecision =
        if (metricKey in preciseKeys) MetricPrecision.PRECISE_DEGREES else MetricPrecision.QUALITATIVE
}

File: poses_viewer/src/drill2d/metricPrecision.ts
/**
 * Trust rule — precise degrees ONLY for the five in-plane metrics; everything else
 * is qualitative-only. 1:1 mirror of Kotlin MetricPrecisionPolicy.
 */
import { METRIC_KEYS } from './referenceStandard'

export type MetricPrecision = 'precise_degrees' | 'qualitative'

const PRECISE = new Set<string>(METRIC_KEYS)

/** Unknown metrics default to qualitative — never overclaim precision. */
export function precisionFor(metricKey: string): MetricPrecision {
  return PRECISE.has(metricKey) ? 'precise_degrees' : 'qualitative'
}
